server/hotrod/src/main/scala/org/infinispan/server/hotrod/HotRodServer.scala

```diff
@@ -247,7 +248,19 @@ class HotRodServer extends AbstractProtocolServer("HotRod") with Log {
       clientListenerRegistry.setEventMarshaller(Option(marshaller))
    }
 
-   override def stop: Unit = {
+   def addCacheStore(name: String, instance: Object): Unit = {
```

Comment:
> Deployed cache stores should be inactive. I can then amend a cache configuration to use that cachestore if needed.

^ I think that might be the most sensible idea, in which case there's no need to fiddle at all with the component registry. The only tricky thing then is to make sure the cache stores/loaders are deployed before caches start, so that any configuration processing can "find" them. Right?